spdif_rx.c

```diff
@@ -404,42 +404,42 @@ int spdif_rx_detect(spdif_rx_samp_freq_t *samp_freq, bool *inverted)
 
     // sampled data analysis to calculate min_width, max_width, max_edge_interval for both 0 and 1
     const int SC = 1; // sample criteria
-    int edge_interval[2] = {0, 0};
-    int min_width[2] = {256, 256};
-    //int max_width[2] = {0, 0}; // not needed for frequency detection
+    int edge_pos[2] = {0, 0}; // unknown
     int max_edge_interval[2] = {0, 0};
+    int min_edge_interval = 0xFFFF;
     {
         //for (int i = 0; i < SPDIF_RX_DETECT_SIZE; i++) printf("data[%3d] = %032b\n", i, fifo_buff[i]);
-        int succ = 0;
-        int cur = 1;
+        int succ = 0; // current bit position in buffer
```

Comment:
sorry for delayed timing review comment. Now I see your code on PC at fist time. Let's change `succ` to proper name such as `pos`, `cycles` or any other which you think is proper name.

Reply:
Done. At some point I had already changed the name, but then kept it out of respect to your existing code, to not "alienate" you.